src/main/java/com/sforce/ws/codegen/metadata/ClassMetadata.java

```diff
@@ -34,19 +34,13 @@ public class ClassMetadata {
     protected final String className;
     protected final String packageName;
     protected final String interfacePackageName;
+    protected final boolean addDeprecatedAnnotation;
 
-    public static ClassMetadata newInstance(String packageName, String className) {
-        return new ClassMetadata(packageName, className);
-    }
-
-    public ClassMetadata(String packageName, String className) {
-        this(packageName, className, null);
-    }
-
-    public ClassMetadata(String packageName, String className, String interfacePackageName) {
+    public ClassMetadata(String packageName, String className, String interfacePackageName, boolean addDeprecatedAnnotation) {
```

Comment:
This change feels a bit more intrusive per say, but anyone who had this before can still use the new way -> new ClassMetadata(packageName, className, null, false) ..